libraries/aws_config_delivery_channel.rb

```diff
@@ -7,46 +7,48 @@ class AwsConfigDeliveryChannel < AwsResourceBase
   desc 'Verifies settings for AWS Config Delivery Channel'
   example "
     describe aws_config_delivery_channel do
-      it { should exist }
+      it                    { should exist }
       its('s3_bucket_name') { should eq 'my_bucket' }
-      its('sns_topic_arn') { should eq arn:aws:sns:us-east-1:721741954427:sns_topic' }
+      its('sns_topic_arn')  { should eq arn:aws:sns:us-east-1:721741954427:sns_topic' }
     end
   "
-  attr_reader :channel_name, :s3_bucket_name, :s3_key_prefix, :sns_topic_arn, :delivery_frequency_in_hours, :exists
-  alias exists? exists
+  attr_reader :channel_name, :s3_bucket_name, :s3_key_prefix, :sns_topic_arn, :delivery_frequency_in_hours
 
   def initialize(opts = {})
-    # Call the parent class constructor
-    opts = { channel_name: opts } if opts.is_a?(String) # this preserves the original scalar interface
+    opts = { channel_name: opts } if opts.is_a?(String)
+
     super(opts)
-    validate_parameters([:channel_name]) if !opts.nil?
-    @channel_name = opts[:channel_name] unless opts.nil?
-    query = @channel_name ? { delivery_channel_names: [@channel_name] } : {}
+    validate_parameters(allow: [:channel_name])
+
+    query = !opts.nil? && opts.key?(:channel_name) ? { delivery_channel_names: [opts[:channel_name]] } : {}
+
     catch_aws_errors do
       begin
-        @resp = @aws.config_client.describe_delivery_channels(query)
-        @exists = !@resp.delivery_channels.empty?
+        resp = @aws.config_client.describe_delivery_channels(query)
       rescue Aws::ConfigService::Errors::NoSuchDeliveryChannelException
-        @exists = false
```

Comment:
This was allowing the `exists` check to work when the exception is raised.  Calling this will likely fail with the changes as `@s3_bucket_name` will be `nil`.

Reply:
Ci Passing on this